### ISSUE TITLE
Error when compiling Application with target wasm32-unknown-unknown

### ISSUE THREAD

**Issue by rafaga** [stainless-steel/sqlite3-src#4]
When i try to run 

`cargo check --workspace --all-features --lib --target wasm32-unknown-unknown`

this happens

```
The following warnings were emitted during compilation:

warning: source/sqlite3.c:14044:10: fatal error: 'stdio.h' file not found
warning: #include <stdio.h>
warning:          ^~~~~~~~~
warning: 1 error generated.

error: failed to run custom build command for `sqlite3-src v0.4.0`

Caused by:
  process didn't exit successfully: `/Users/rafael/projects/telescope/target/debug/build/sqlite3-src-fe63e4ca8bda58d7/build-script-build` (exit status: 1)
  --- stdout
  cargo:rerun-if-env-changed=SQLITE3_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS_wasm32-unknown-unknown
  cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS_wasm32_unknown_unknown
  cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_ALLOW_CROSS
  cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS
  cargo:rerun-if-env-changed=PKG_CONFIG_wasm32-unknown-unknown
  cargo:rerun-if-env-changed=PKG_CONFIG_wasm32_unknown_unknown
  cargo:rerun-if-env-changed=TARGET_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_wasm32-unknown-unknown
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_wasm32_unknown_unknown
  cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  TARGET = Some("wasm32-unknown-unknown")
  OPT_LEVEL = Some("2")
  HOST = Some("x86_64-apple-darwin")
  cargo:rerun-if-env-changed=CC_wasm32-unknown-unknown
  CC_wasm32-unknown-unknown = None
  cargo:rerun-if-env-changed=CC_wasm32_unknown_unknown
  CC_wasm32_unknown_unknown = None
  cargo:rerun-if-env-changed=TARGET_CC
  TARGET_CC = None
  cargo:rerun-if-env-changed=CC
  CC = None
  cargo:rerun-if-env-changed=CFLAGS_wasm32-unknown-unknown
  CFLAGS_wasm32-unknown-unknown = None
  cargo:rerun-if-env-changed=CFLAGS_wasm32_unknown_unknown
  CFLAGS_wasm32_unknown_unknown = None
  cargo:rerun-if-env-changed=TARGET_CFLAGS
  TARGET_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("true")
  running: "clang" "-O2" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "--target=wasm32-unknown-unknown" "-Wall" "-Wextra" "-o" "/Users/rafael/projects/telescope/target/wasm32-unknown-unknown/debug/build/sqlite3-src-5852d0769e37260f/out/source/sqlite3.o" "-c" "source/sqlite3.c"
  cargo:warning=source/sqlite3.c:14044:10: fatal error: 'stdio.h' file not found
  cargo:warning=#include <stdio.h>
  cargo:warning=         ^~~~~~~~~
  cargo:warning=1 error generated.
  exit status: 1

  --- stderr


  error occurred: Command "clang" "-O2" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "--target=wasm32-unknown-unknown" "-Wall" "-Wextra" "-o" "/Users/rafael/projects/telescope/target/wasm32-unknown-unknown/debug/build/sqlite3-src-5852d0769e37260f/out/source/sqlite3.o" "-c" "source/sqlite3.c" with args "clang" did not execute successfully (status code exit status: 1).


warning: build failed, waiting for other jobs to finish...
```

I'm running with rust 1.67.1 on MacOS 13.2

**Comment by ptechen** [comment]
Me, too!  Did you solve it? Is it convenient to let me know?

**Comment by IvanUkhov** [comment]
Does SQLite support this target in general? If you know how to compile standalone, that is, independently of this crate, it would be a matter of updating `build.rs`.

**Comment by rafaga** [comment]
what I understood from a conversation in the Discord channel is that this issue it is related with a ABI compatibility problem within rust - C lang, beyond the scope of this crate ... so it is not feasible for the moment to resolve this issue ... The rust team it's addressing the problem , but no ETA provided

**Comment by srussvoll** [comment]
We get this error because libc does not work on `wasm32-unknown-unknown`, which in turn is because the target does not specify any interfaces for libc to use. WASI is a suitable interface that libc supports, and SQLite has had support for compiling for `wasm32-wasip1` since version 3.41.0. Today it also supports `wasm32-wasip2`. I added a PR to add support for this that I have tested in the browser.

- #8
